wgengine/netstack/netstack.go

```diff
@@ -147,10 +147,10 @@ type Impl struct {
 
 const nicID = 1
 
-// maxUDPPacketSize is the maximum size of a UDP packet we copy in startPacketCopy
-// when relaying UDP packets. We don't use the 'mtu' const in anticipation of
-// one day making the MTU more dynamic.
-const maxUDPPacketSize = 1500
+// maxUDPPacketSize is the maximum size of a UDP packet we copy in
+// startPacketCopy when relaying UDP packets. We support arbitrarily
+// large MTUs so this is now the maximum IP packet size.
+const maxUDPPacketSize = packet.MaxIPPacketSize
```

Comment:
I don't fully understand how hot `startPacketCopy` is, but it allocates a buffer of this size, which is now 43 times larger. It seems like we can safely use `MaxProbedWireMTU` instead?

Reply:
Thanks, I didn't like this either. I think what we actually want is the maximum of MaxProbedWireMTU and DefaultTunMTU + wg overhead (due to user setting of MTU via ifconfig or TS_DEBUG_MTU).

Reply:
Oof. I still don't fully understand this code, but I do believe that that the user can configure the tailscale MTU up to tstun.MaxPacketSize and we have to be able to handle that.